src/lightning/data/streaming/reader.py

```diff
@@ -294,16 +291,16 @@ def _get_folder_size(path: str) -> int:
     return size
 
 
-def _get_from_queue(queue: multiprocessing.Queue) -> Optional[Any]:
+def _get_from_queue(queue: multiprocessing.Queue, timeout: float = _DEFAULT_TIMEOUT) -> Optional[Any]:
     try:
         # Note: The timeout here should not be too short. We need to prevent the caller from aggressively
-        #   querying the queue and consuming too many CPU cycles.
-        return queue.get(timeout=0.1)
+        # querying the queue and consuming too many CPU cycles.
+        return queue.get(timeout=timeout)
     except Empty:
         pass
     except OSError as e:
         # handle closed queue before the thread terminates
-        if "handle is closed" in str(e):
+        if "handle is closed" in str(e) or "Bad file descriptor" in str(e):
```

Comment:
```suggestion
        if "handle is closed" in str(e):
```
(this was originally my debug branch before you opened the PR on it)

Reply:
removed.